components/eventing-controller/controllers/subscriptionv2/jetstream/reconciler.go

```diff
@@ -197,81 +184,121 @@ func (r *Reconciler) enqueueReconciliationForSubscriptions(subs []eventingv1alph
-			events.Warn(r.recorder, sub, events.ReasonUpdateFailed, "Update Subscription status failed %s", sub.Name)
-			return pkgerrors.MakeError(errFailedToUpdateStatus, updateErr)
-		}
-		events.Normal(r.recorder, sub, events.ReasonUpdate, "Update Subscription status succeeded %s", sub.Name)
-	}
-	return nil
-}
-
 // handleSubscriptionDeletion deletes the JetStream subscription and removes its finalizer if it is set.
 func (r *Reconciler) handleSubscriptionDeletion(ctx context.Context,
-	subscription *eventingv1alpha2.Subscription, log *zap.SugaredLogger) error {
+	subscription *eventingv1alpha2.Subscription, log *zap.SugaredLogger) (ctrl.Result, error) {
+	// delete the JetStream subscription/consumer
 	if utils.ContainsString(subscription.ObjectMeta.Finalizers, eventingv1alpha2.Finalizer) {
 		if err := r.Backend.DeleteSubscription(subscription); err != nil {
 			// if failed to delete the external dependency here, return with error
 			// so that it can be retried
-			return pkgerrors.MakeError(errFailedToDeleteSub, err)
+			if syncErr := r.syncSubscriptionStatus(ctx, subscription, err, log); syncErr != nil {
+				return ctrl.Result{}, syncErr
+			}
+			return ctrl.Result{}, pkgerrors.MakeError(errFailedToDeleteSub, err)
 		}
 
-		// remove our finalizer from the list and update it.
+		// remove the eventing finalizer from the list and update the subscription.
 		subscription.ObjectMeta.Finalizers = utils.RemoveString(subscription.ObjectMeta.Finalizers,
 			eventingv1alpha2.Finalizer)
-		if err := r.Client.Update(ctx, subscription); err != nil {
-			events.Warn(r.recorder, subscription, events.ReasonUpdateFailed, "Update Subscription failed %s", subscription.Name)
-			return pkgerrors.MakeError(errFailedToRemoveFinalizer, err)
+		return ctrl.Result{}, r.syncSubscriptionStatus(ctx, subscription, nil, log)
+	}
+	return ctrl.Result{}, nil
+}
+
+// syncSubscriptionStatus syncs Subscription status and updates the k8s subscription.
+func (r *Reconciler) syncSubscriptionStatus(ctx context.Context,
+	desiredSubscription *eventingv1alpha2.Subscription, err error, log *zap.SugaredLogger) error {
+	// set ready state
+	desiredSubscription.Status.Ready = err == nil
+
+	// compile the desired conditions
+	desiredSubscription.Status.Conditions = eventingv1alpha2.GetSubscriptionActiveCondition(desiredSubscription, err)
+
+	// Update the subscription
+	return r.updateSubscription(ctx, desiredSubscription, log)
+}
+
+// updateSubscription updates the subscription changes to k8s.
+func (r *Reconciler) updateSubscription(ctx context.Context,
+	sub *eventingv1alpha2.Subscription, logger *zap.SugaredLogger) error {
+	namespacedName := &k8stypes.NamespacedName{
+		Name:      sub.Name,
+		Namespace: sub.Namespace,
+	}
+
+	// fetch the latest subscription object, to avoid k8s conflict errors
+	actualSubscription := &eventingv1alpha2.Subscription{}
+	if err := r.Client.Get(ctx, *namespacedName, actualSubscription); err != nil {
+		return err
+	}
```

Comment:
This is added on purpose. This is following the same pattern as the eventMesh reconciler. I think there was discovered, that sometimes the during the reconciliation the resource version might change, so that the `currentSub` won't be the actual subscription anymore. That is the reason of this code block.

Reply:
This is the same approach we follow in the new EventMesh reconciler. The problem there is, that sometimes during the reconciliation request is being processes, the ResourceVersion could change in the background by other concurrent reconciliation request. That would lead that our "currentSub" wouldn't be actual anymore and would lead to some errors with conflicting resourceVersions. That's the reason for this request.